selfdrive/thermald/tests/test_power_monitoring.py

```diff
@@ -177,6 +179,31 @@ def test_harness_connection(self):
         if i % 10 == 0:
           self.assertFalse(pm.should_shutdown(ignition, False, ssb, False))
       self.assertFalse(pm.should_shutdown(ignition, False, ssb, False))
+      
+  def test_delay_shutdown_time(self):
+    DELAY_SHUTDOWN_TIME_S = 300
+    with pm_patch("HARDWARE.get_current_power_draw", 0), \
+         pm_patch("DELAY_SHUTDOWN_TIME_S", DELAY_SHUTDOWN_TIME_S, constant=True):
+           
+      pm = PowerMonitoring()
+      pm.car_battery_capacity_uWh = 0
+      ignition = False
+      in_car = True
+      start_time = ssb  
+      offroad_timestamp = start_time
+      started_seen = False
+  
+      while start_time <= offroad_timestamp + DELAY_SHUTDOWN_TIME_S:
+        pm.calculate(VOLTAGE_BELOW_PAUSE_CHARGING, ignition)
+        should_shutdown = pm.should_shutdown(ignition, in_car, offroad_timestamp, started_seen)
+        self.assertFalse(should_shutdown, f"Should not shutdown before {DELAY_SHUTDOWN_TIME_S} seconds offroad time")
+        start_time += 1  # Simulate the passage of time
+      started_seen = True
```

Comment:
`started_seen` should be high for the whole test period, otherwise the shutdown delay logic is completely masked by it!

Reply:
Yea I realized this afterwards.